probably/pgcl/syntax.py

```diff
@@ -153,6 +157,8 @@ def check_is_linear_expr(context: Optional[Program], expr: Expr) -> Optional[Che
     """
 
     def _has_variable(expr: Expr) -> bool:
+        if isinstance(expr, VarExpr) and context is not None and expr.var in context.constants:
+            raise Exception("The context program must not contain constants")
```

Comment:
This exception message is misleading: It should say that the expression must not contain constants. The context may of course declare constants.